pywbem/cim_operations.py

```diff
@@ -643,6 +671,24 @@ def __init__(self, url, creds=None, default_namespace=DEFAULT_NAMESPACE,
             Note that not all situations can be handled within this timeout, so
             for some issues, operations may take longer before raising an
             exception.
+
+          use_pull_operations (:class:`py:bool`):
+            Controls the use of pull operations in any `Iter...()` methods.
+            The default value (`None`) will work on any WBEM server whether or
+            not it supports pull operations so that first the pull operation is
+            attempted and if the WBEM server does not support pull operations,
+            the corresponding instance variable is set to `False` the
+            traditional (non-pull) operation will be executed for the current
+            operation and all subsequent  Iter...()operations for this
+            WBEM Connection.
+
+            `True` means that pull operations will be attempted; all errors
+            including `CIM_ERR_NOT_SUPPORTED` are returned to the caller. The
+            traditional operations will not be attempted.
```

Comment:
No.  The way the code works is that if True is set on the attribute, a failure of the open or pull goes directly to raise in all cases.  The assumption we are making is that if true is set either by the attribute or initial success, all operations MUST be pull.   Unless I misread my code that is what it does.  The only time it first tries pull and then falls back to enum is if use... is None and the error is CIM_ERR_NOT_SUPPORTED. 

                except CIMError as ce:
                    if self.use_pull_operations is None and \
                       ce.status_code == CIM_ERR_NOT_SUPPORTED:
                        self.use_pull_operations = False
                    else:
                        raise

Any other exception gets raised directly.